### ISSUE TITLE
I have a quick and a simple question about saving for each checkpoint, each epoch training model.

### ISSUE THREAD

**Issue by star4s** [nickyisadog/latent-diffusion-inpainting#15]
Hello, I am so sorry because of bothering on several time.
I do not have account wechat, so I cannot contact to you.
Maybe next, I will try making a wechat account.

I have a question about saving for each checkpoint, each epoch training model.
After I run "CUDA_VISIBLE_DEVICES=0 python main.py --base ldm/models/ldm/inpainting_big/config.yaml --stage 1 -t --gpus 0, ",
it just overwrite to save "last.ckpt" in the folder "logs\checkpoints\".
So. I checked the main code, there is "'save_top_k': -1" in default_metrics_over_trainsteps_ckpt_dict().

=========================

        if 'metrics_over_trainsteps_checkpoint' in callbacks_cfg:
            print(
                'Caution: Saving checkpoints every n train steps without deleting. This might require some free space.')
            default_metrics_over_trainsteps_ckpt_dict = {
                'metrics_over_trainsteps_checkpoint':
                    {"target": 'pytorch_lightning.callbacks.ModelCheckpoint',
                     'params': {
                         "dirpath": os.path.join(ckptdir, 'trainstep_checkpoints'),
                         "filename": "{epoch:06}-{step:09}",
                         "verbose": True,
                         'save_top_k': -1,
                         'every_n_train_steps': 10000,
                         'save_weights_only': True
                     }
                     }
            }
            default_callbacks_cfg.update(default_metrics_over_trainsteps_ckpt_dict)

==========================================

However, It just overwrite to save "last.ckpt".
How can I save each epoch checkpoint?

I have some warning massage:

.conda/envs/ldm2/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:216: UserWarning: Please also save or load the state of the optimizer when saving or loading the scheduler.
  warnings.warn(SAVE_STATE_WARNING, UserWarning)

In your case, did you have every epoch checkpoint?

In addition, did you set validation data for training?

Did you modify the file, "\ldm\models\ldm\inpainting_big\config.yaml"?

in the config.yalm,

data:
  target: main.DataModuleFromConfig
  params:
    batch_size: 1
    wrap: true
    train:
      target: "/ldm/Train"
      params:
        size: 512
        crop_size: 256
    validation:
      target: "/ldm/Val"
      params:
        size: 512
        crop_size: 256

Thank you for your attention and help.

**Comment by Srutarshi** [comment]
I have the same problem. Did you have a solution for this ?

**Comment by ultiwinter** [comment]
@star4s hello, thanks for putting the question out there. did you find the solution?
